### PR TITLE
Use Go Modules

### PR DESCRIPTION
**Pull request by SamWhited**
Per a comment from @thaJeztah, let's consider using Go Modules

### CODE REVIEW

**Comment by SamWhited** [comment]
I can't tell if tests actually ran or not, it wants me to log in but I'm not sure if that means the test runner needs someone to auth to set it up, or it just wants me to log in before I can see results.

**Comment by thaJeztah** [comment]
```
#!/bin/bash -eo pipefail
go get -u golang.org/x/lint/golint &&
/bin/bash: -c: line 1: syntax error: unexpected end of file

Exited with code exit status 2
```

File: circle.yml [hunk 1]
```diff
@@ -12,9 +11,6 @@ jobs:
           name: Get dependencies
           command: |
             go get -u golang.org/x/lint/golint &&
```
[the remark this answers is not in the document]
Comment (SamWhited) [reply circle.yml]: Oops, fixed. Is it possible to make the CI not require login?

[the remark this answers is not in the document]
Comment (thaJeztah) [reply circle.yml]: No idea tbh, I usually authenticate with GitHub and that seems to work (also on other repositories)

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,9 @@
+module github.com/docker/go-connections
+
+go 1.13
+
+require (
+	github.com/Microsoft/go-winio v0.4.14
+	github.com/pkg/errors v0.9.1
+	github.com/stretchr/testify v1.6.0
```
[the remark this answers is not in the document]
Comment (thaJeztah) [reply go.mod]: Sometimes `<insert assert framework>` is useful, but in this case it was just used in three tests, so thought it was worth looking at; thanks!

**Comment by thaJeztah** [comment]
This is weird;

```
#!/bin/bash -eo pipefail
go build ./...
tlsconfig/config.go:16:2: cannot find package "github.com/pkg/errors" in any of:
	/usr/local/go/src/github.com/pkg/errors (from $GOROOT)
	/home/circleci/.go_workspace/src/github.com/pkg/errors (from $GOPATH)
	/usr/local/go_workspace/src/github.com/pkg/errors

Exited with code exit status 1
```

**Comment by SamWhited** [comment]
It must be using an older version of Go (so no Go Modules by default). I don't see a way to change it from quickly glancing around the interwebs (everywhere seems to use a different builder than we do, but that seems like a lot more work and risk to change, especially when I can't actually see the CI output). I'll keep digging around and see what I can find.

File: circle.yml [hunk 3]
```diff
@@ -2,7 +2,6 @@ version: 2
 jobs:
   build:
     machine: true
-    working_directory: ~/go/src/github.com/docker/go-connections
```
Comment (thaJeztah) [inline circle.yml:5]: Can you try adding (from the list of https://circleci.com/docs/2.0/configuration-reference/#available-machine-images)

```suggestion
    image: ubuntu-1604:202004-01
```

Comment (SamWhited) [reply circle.yml:5]: Oops, got distracted by work and forgot about this one. Yup, I'll give it a shot.

Comment (thaJeztah) [reply circle.yml:5]: Argh. Think I messed up indentation;

```
# Unable to parse YAML
# mapping values are not allowed here
#  in 'string', line 5, column 12:
#           image: ubuntu-1604:202004-01
#                ^
# 
# -------
# Warning: This configuration was auto-generated to show you the message above.
# Don't rerun this job. Rerunning will have no effect.
false
```

Comment (SamWhited) [reply circle.yml:5]: I hate YAML… sorry, I even noticed that my editor kept trying to move that around, should have listened to it.

Comment (thaJeztah) [reply circle.yml:5]: Yes, looks like the "boolean" may be an old variant, and it needs to be; https://circleci.com/docs/2.0/configuration-reference/#machine

```yaml
jobs:
  build:
    machine:
      image: ubuntu-1604:202004-01
```

**Comment by thaJeztah** [comment]
@vdemeester @AkihiroSuda PTAL 🤗

**Comment by AkihiroSuda** [comment]
Needs sign? (`git commit -s`)

**Comment by thaJeztah** [comment]
> Needs sign? (git commit -s)

arf, good catch; looks like there's (no longer?) a DCO check running in this repo

**Comment by SamWhited** [comment]
Fixed, thanks.

**Comment by thaJeztah** [comment]
Whoop! DCO bot is also working now 👍

**Pull request merged**